File: java_solutions/Difficulty1.4/Parking.java
// Program by Muhammad Bagus Zulmi
// Url to problem : https://open.kattis.com/problems/parking2
// http://www.mbaguszulmi.com

import java.util.Scanner;

public class Parking {
    public static int[] convertToOrderedIntegerArray(String[] pos) {
        int[] intArray = new int[pos.length];
        for (int i = 0; i < pos.length; i++) {
            intArray[i] = Integer.parseInt(pos[i]);
        }

        for (int i = 0; i < intArray.length-1; i++) {
            int idx = i;
            for (int j = i+1; j < intArray.length; j++) {
                if (intArray[idx] > intArray[j]) {
                    idx = j;
                }
            }
            int temp = intArray[i];
            intArray[i] = intArray[idx];
            intArray[idx] = temp;
        }
        return intArray;
    }

    public static int optimal(int[] intArray) {
        int distance=0;
        for (int i = 1; i < intArray.length; i++) {
            distance+=intArray[i]-intArray[i-1];
        }
        return distance*2;
    }

    public static void main(String[] args) {
        Scanner userIn = new Scanner(System.in);
        int t = userIn.nextInt();
        userIn.nextLine();
        int[] optimalDistance = new int[t];
        for (int i = 0; i < t; i++) {
            userIn.nextInt();
            userIn.nextLine();
            String[] input = userIn.nextLine().split(" ");
            optimalDistance[i] = optimal(convertToOrderedIntegerArray(input));
        }

        for (int i = 0; i < t; i++) {
            System.out.println(optimalDistance[i]);
        }
        userIn.close();
    }
}

File: python27_solutions/Difficulty1.5/Parking.py
# Program by Muhammad Bagus Zulmi
# Url to problem : https://open.kattis.com/problems/parking
# http://www.mbaguszulmi.com

def main():
	inputStr = raw_input().split()
	cost = [int(inputStr[0]), int(inputStr[1]), int(inputStr[2])]

	totalCost = 0
	times = []
	_min = 101
	_max = 0
	for i in range(3):
		inputStr = raw_input().split()
		arrival = int(inputStr[0])
		departure = int(inputStr[1])
		times.append(range(arrival, departure))
		if arrival < _min:
			_min = arrival
		
		if departure > _max:
			_max = departure
	
	for i in range(_min, _max):
		numCar = 0
		for j in range(3):
			if i in times[j]:
				numCar+=1
		
		totalCost+=numCar*cost[numCar-1]
	
	print totalCost

if __name__ == '__main__':
	main()